### PR TITLE
SM-948: Update vehicle-signal-decoding to accept different auth mechanism

### PR DESCRIPTION
**Pull request by zakharenkodmytro**
- added EC recover authentication as second auth to "/device-config/eth-addr/:ethAddr/status" PATCH api
Screenshot 2024-07-02 at 2 27 05 PM

### CODE REVIEW

**Comment by zakharenkodmytro** [comment]
This PR enables two ways to authenticate to the fiber endpoints: JWT auth and EcRecover auth. If JWT authentication fails, we call the next middleware function, EC recover authentication. EC recover could fail because of an internal Server Error or because VerifySignature failed and the Unauthorized status code returned. If EC recover authentication succeeds, we call the next authorization middleware function.

File: internal/api/api.go [hunk 1]
```diff
@@ -144,8 +145,35 @@ func startWebAPI(logger zerolog.Logger, settings *config.Settings, database db.S
 		ErrorHandler: func(_ *fiber.Ctx, err error) error {
 			return fiber.NewError(fiber.StatusUnauthorized, "Invalid JWT. "+err.Error())
 		},
+		SuccessHandler: func(_ *fiber.Ctx) error {
```
Comment (JamesReate) [inline internal/api/api.go:148]: just want to check this will still work with regular auth JWT - we can test in dev

Comment (zakharenkodmytro) [reply internal/api/api.go:148]: Let's test it.

File: internal/test/helpers.go [hunk 2]
```diff
@@ -122,6 +122,11 @@ type UsersClient struct {
 	Store map[string]*pb.User
 }
 
+func (c *UsersClient) GetUserByEthAddr(_ context.Context, _ *pb.GetUserByEthRequest, _ ...grpc.CallOption) (*pb.User, error) {
+	//TODO implement me
+	panic("implement me")
```
Comment (JamesReate) [inline internal/test/helpers.go:127]: is this ok like this, or just so interface contract works?

**Review by JamesReate — APPROVED** [review]
lgtm, but let's test it in dev before going forward. 
is the utils/crypto just copied from devices-api? wonder if candidate for shared library.

File: internal/test/helpers.go (hunk 2 above)
Comment (zakharenkodmytro) [reply internal/test/helpers.go:127]: yeah, this is so interface contracts work.

**Comment by zakharenkodmytro** [comment]
> lgtm, but let's test it in dev before going forward. is the utils/crypto just copied from devices-api? wonder if candidate for shared library.

>
we could extract it to the shared

File: internal/test/helpers.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (zakharenkodmytro) [reply internal/test/helpers.go:127]: good point, I just quickly add it so interface contract would work

**Review by Asfiroth — APPROVED** [review]

**Pull request merged**